metrics_ci_integration/lib/jenkins/client/model/jenkins_query_limits.dart

```diff
@@ -0,0 +1,183 @@
+  ///
+  /// Specifies the `{number}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.at(3)` stands for the third element.
+  factory JenkinsQueryLimits.at(int number) {
+    NumberValidator.checkPositive(number);
+    return JenkinsQueryLimits._(lower: number, upper: number);
+  }
+
+  /// Creates a range specifier for the left-edged range (exclusive).
+  ///
+  /// Specifies the `{number+1,}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.startAfter(3)` stands for the range from the third
+  ///     element (exclusive) to the end.
+  factory JenkinsQueryLimits.startAfter(int number) {
+    NumberValidator.checkPositive(number);
+    return JenkinsQueryLimits._(lower: number + 1);
+  }
+
+  /// Creates a range specifier for the left-edged range (inclusive).
+  ///
+  /// Specifies the `{number,}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.startAt(3)` stands for the range from the third
+  ///     element (inclusive) to the end.
+  factory JenkinsQueryLimits.startAt(int number) {
+    NumberValidator.checkPositive(number);
+    return JenkinsQueryLimits._(lower: number);
+  }
+
+  /// Creates a range specifier for the right-edged range (exclusive).
+  ///
+  /// Specifies the `{,number}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.endBefore(3)` stands for the range from the begin
+  ///     to the third element (exclusive).
+  factory JenkinsQueryLimits.endBefore(int number) {
+    NumberValidator.checkPositive(number);
+    return JenkinsQueryLimits._(upper: number);
+  }
+
+  /// Creates a range specifier for the right-edged range (inclusive).
+  ///
+  /// Specifies the `{,number+1}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.endAt(3)` stands for the range from the begin
+  ///     to the third element (inclusive).
+  factory JenkinsQueryLimits.endAt(int number) {
+    NumberValidator.checkPositive(number);
+    return JenkinsQueryLimits._(upper: number + 1);
+  }
+
+  /// Creates a range specifier for the left-edged (inclusive) and
+  /// right-edged (exclusive) range.
+  ///
+  /// Specifies the `{begin,end}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.between(3, 10)` stands for the range from the
+  ///     third element (inclusive) to the tenth element (exclusive).
+  factory JenkinsQueryLimits.between(int begin, int end) {
+    NumberValidator.checkRange(begin, end);
+    return JenkinsQueryLimits._(lower: begin, upper: end);
+  }
+
+  /// Creates a range specifier for the left-edged (inclusive) and
+  /// right-edged (inclusive) range.
+  ///
+  /// Specifies the `{begin,end+1}` range.
+  /// For example:
+  ///   * `JenkinsQueryLimits.betweenInclusive(3, 10)` stands for the range from
+  ///     the third element (inclusive) to the tenth element (inclusive).
+  factory JenkinsQueryLimits.betweenInclusive(int begin, int end) {
+    NumberValidator.checkRange(begin, end);
+    return JenkinsQueryLimits._(lower: begin, upper: end + 1);
+  }
```

Comment:
Do we really need such "Inclusive" methods? I think we can always use simple "between" and adjust parameters to get expected results. Maybe we should just remove them?